src/components/TodoDetails.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (todoId: number) => void,
+  isActive: boolean,
+  deletedTodoIds: number[],
+  activeTodoIds: number[],
+  updateTodoData: (todoId: number, data: object) => void,
+};
+
+export const TodoDetails: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  isActive,
+  deletedTodoIds,
+  activeTodoIds,
+  updateTodoData,
+}) => {
+  const [isRenamingTodo, setIsRenamingTodo] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  });
+
+  const isDeleted = (todoId: number) => {
+    return deletedTodoIds?.includes(todoId);
+  };
+
+  const isLoadingTodo = (todoId: number) => {
+    if (deletedTodoIds.length !== 0) {
```

Comment:
I guess you can handle `loading` circle rendering with only one state property, like `loadingTodoIds` - and push/remove (at the start/end of action) there ids on any action (deleting/adding/editing), since you can do only one action at the time.

It'd simplify the app.
But it's just FYI - anyway - GJ :+1: